components/schemes/Container/src/MDCContainerScheme.h

```diff
@@ -0,0 +1,43 @@
+// Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <Foundation/Foundation.h>
+
+#import "MaterialColorScheme.h"
+#import "MaterialShapeScheme.h"
+#import "MaterialTypographyScheme.h"
+
+@protocol MDCContainerScheming <NSObject>
+
+@property(nonatomic, nullable, readonly) id<MDCColorScheming> colorScheme;
+@property(nonatomic, nullable, readonly) id<MDCTypographyScheming> typographyScheme;
+@property(nonatomic, nullable, readonly) id<MDCShapeScheming> shapeScheme;
+
+@end
+
+typedef NS_ENUM(NSInteger, MDCContainerSchemeDefaults) {
+  MDCContainerSchemeDefaultsMaterial201811,
+};
+
+__attribute__((objc_subclassing_restricted)) @interface MDCContainerScheme
+    : NSObject<MDCContainerScheming>
+
+@property(nonatomic, nullable) id<MDCColorScheming> colorScheme;
+@property(nonatomic, nullable) id<MDCTypographyScheming> typographyScheme;
+@property(nonatomic, nullable) id<MDCShapeScheming> shapeScheme;
```

Comment:
Each of these properties should be the mutable version of their corresponding scheming type.

E.g. MDCSemanticColorScheme.

Reply:
I think they are the mutable version of the corresponding scheming type, by default the property is `readwrite`? I set the scheming type is `id<MDCColorScheming>`. Though the doc stated the `MDCSemanticColorScheme`,  I wanted to discussed on it when reading the doc.  I feel `id<MDCColorScheming>` would be more extensive in this case, especially when we don't allow subclassing in the subsystem scheme. What do you think?

Reply:
Implementations don't need to inherit from each other, only the protocols. I think it's simpler for the implementation to stick to concrete implementations of each subsystem scheme.  By using the protocols, we lose mutability of the Container Scheme overall.  Either way, I believe the decision was made to stick with the concrete types in the Container implementation.

Reply:
Sure. Addressed in d1e1751.